turf/src/commonMain/kotlin/io/github/elcolto/geokjson/turf/booleans/Within.kt

```diff
@@ -0,0 +1,226 @@
+        if (!pointInPolygon(position, polygon)) return false
+        foundInsidePoint = foundInsidePoint || pointInPolygon(position, polygon, ignoreBoundary = true)
+    }
+    return foundInsidePoint
+}
+
+@OptIn(ExperimentalTurfApi::class)
+private fun isMultiPointInPoly(multiPoint: MultiPoint, polygon: MultiPolygon): Boolean {
+    var foundInsidePoint = false
+    multiPoint.coordinates.forEach { position ->
+        if (!pointInPolygon(position, polygon)) return false
+        foundInsidePoint = foundInsidePoint || pointInPolygon(position, polygon, ignoreBoundary = true)
+    }
+    return foundInsidePoint
+}
+
+@OptIn(ExperimentalTurfApi::class)
+private fun isLineOnLine(lineString1: LineString, lineString2: LineString) =
+    lineString1.coordinates.all { position -> pointOnLine(position, lineString2) }
+
+@OptIn(ExperimentalTurfApi::class)
+private fun isLineInPoly(linestring: LineString, polygon: Polygon): Boolean {
+    val polyBbox = polygon.bbox ?: computeBbox(polygon.coordAll())
+    val lineBbox = linestring.bbox ?: computeBbox(linestring.coordAll())
+
+    if (!doBBoxOverlap(polyBbox, lineBbox)) {
+        return false
+    }
+
+    return isLineInPoly(linestring.coordinates, polygon)
+}
+
+@OptIn(ExperimentalTurfApi::class)
+private fun isLineInPoly(coordinates: List<Position>, polygon: Polygon): Boolean {
+    var foundInsidePoint = false
+
+    coordinates.forEachIndexed { index, position ->
+        if (!pointInPolygon(position, polygon)) return false
+
+        foundInsidePoint = foundInsidePoint || pointInPolygon(position, polygon, ignoreBoundary = true)
+
+        if (!foundInsidePoint && index < coordinates.lastIndex) {
+            val midpoint = midpoint(position, coordinates[index + 1])
+            foundInsidePoint = foundInsidePoint || pointInPolygon(midpoint, polygon, ignoreBoundary = true)
+        }
+    }
+
+    return foundInsidePoint
+}
+
+@OptIn(ExperimentalTurfApi::class)
+private fun isLineInPoly(linestring: LineString, polygon: MultiPolygon): Boolean {
+    val polyBbox = polygon.bbox ?: computeBbox(polygon.coordAll())
+    val lineBbox = linestring.bbox ?: computeBbox(linestring.coordAll())
+
+    if (!doBBoxOverlap(polyBbox, lineBbox)) {
+        return false
+    }
+
+    return isLineInPoly(linestring.coordinates, polygon)
+}
+
+@OptIn(ExperimentalTurfApi::class)
+private fun isLineInPoly(coordinates: List<Position>, polygon: MultiPolygon): Boolean {
+    var foundInsidePoint = false
+
+    coordinates.forEachIndexed { index, position ->
+        if (!pointInPolygon(position, polygon)) return false
+
+        foundInsidePoint = foundInsidePoint || pointInPolygon(position, polygon, ignoreBoundary = true)
+
+        if (!foundInsidePoint && index < coordinates.lastIndex) {
+            val midpoint = midpoint(position, coordinates[index + 1])
+            foundInsidePoint = foundInsidePoint || pointInPolygon(midpoint, polygon, ignoreBoundary = true)
+        }
+    }
+
+    return foundInsidePoint
```

Comment:
I think gemini didn't understand something xD. The function above calls this one. Calling it would require a LineString, but coordinates is just a list of Positions :joy: And even if we could call it we would run into recursion :grimacing: 
I just joined these two functions with the previous approach